source/fe/fe_system.cc

```diff
@@ -2251,64 +2251,71 @@ FESystem<dim,spacedim>::get_constant_modes () const
 void
 FESystem<dim,spacedim>::
-convert_generalized_support_point_values_to_nodal_values (const std::vector<Vector<double> > &support_point_values,
-                                                          std::vector<double>                &nodal_values) const
+convert_generalized_support_point_values_to_dof_values (const std::vector<Vector<double> > &point_values,
+                                                        std::vector<double>                &dof_values) const
 {
-  // we currently only support the case where each base element has exactly
-  // as many generalized support points as there are dofs in that element.
-  //
-  // in the more general case, if there are more generalized support
-  // points than dofs, we don't have the infrastructure at the time of
-  // writing this to tease apart which support point value
-  // belongs to which element
-  AssertDimension (support_point_values.size(), this->dofs_per_cell);
-  AssertDimension (nodal_values.size(), this->dofs_per_cell);
-
-  // loop over the bases and let them do the work on their
-  // share of the input argument
-  std::vector<Vector<double>> base_support_point_values;
-  std::vector<double>         base_nodal_values;
-  unsigned int                current_vector_component = 0;
-  for (unsigned int base=0; base<base_elements.size(); ++base)
+  Assert(this->has_generalized_support_points(),
+         ExcMessage("The FESystem does not have generalized support points"));
+
+  AssertDimension(point_values.size(),
+                  this->get_generalized_support_points().size());
+  AssertDimension(dof_values.size(), this->dofs_per_cell);
+
+  // loop over all base elements (respecting multiplicity) and let them do
+  // the work on their share of the input argument
+
+  unsigned int current_vector_component = 0;
+  for (unsigned int base = 0; base < base_elements.size(); ++base)
     {
-      const unsigned int element_multiplicity = this->element_multiplicity(base);
-      for (unsigned int m=0;
-           m<element_multiplicity;
-           ++m, current_vector_component += base_element(base).n_components())
+      // We need access to the base_element, its multiplicity, the
+      // number of generalized support points (n_base_points) and the
+      // number of components we're dealing with.
+      const auto &base_element = this->base_element(base);
+      const unsigned int multiplicity = this->element_multiplicity(base);
+      const unsigned int n_base_dofs = base_element.dofs_per_cell;
+      const unsigned int n_base_points =
+        base_element.get_generalized_support_points().size();
+      const unsigned int n_base_components = base_element.n_components();
+
+      // FIXME: These temporary vectors are extremely unfortunate. But there
+      // is no easy way around copying all data to make a recursive call...
+      std::vector<double> base_dof_values(n_base_dofs);
+      std::vector<Vector<double> > base_point_values(n_base_points);
```

Comment:
This function is really not called very frequently, so I wouldn't worry about the FIXME. But if you do care, move the declaration of the vector outside the loop and just do a `resize` here.

Reply:
The problem is the recursive call that might allocate this vector again. But let me simply remove the FIXME.

A warning: This function will be called significantly more often in the future, we plan to reimplement VectorTools::interpolate to exclusively use generalized support points.

I suggest we do some benchmarking after that is done and optimize if necessary.

Reply:
That's ok with me. But if you're concerned, you can still move the declaration outside the loop -- this way, at least you don't have to re-allocate memory all the time.